### PR TITLE
some minor improvements (no logic change)

### PR DESCRIPTION
**Pull request by anuriq**
- remove unused functions and variables
- put pointers in sync.Pool for http input plugin (better performance)

### CODE REVIEW

File: pipeline/pipeline.go [hunk 1]
```diff
@@ -273,7 +273,7 @@ func (p *Pipeline) In(sourceID SourceID, sourceName string, offset int64, bytes
 	}
 
 	event := p.eventPool.get()
-	dec := decoder.NO
+	var dec decoder.DecoderType
```
Comment (vano144) [inline pipeline/pipeline.go:276]: Does the previous version look more readable?

**Comment by vano144** [comment]
LGTM

File: pipeline/pipeline.go (hunk 1 above)
Comment (anuriq) [reply pipeline/pipeline.go:276]: linter doesn't like it - ineffective assignment

**Comment by ansakharov** [comment]
Good job!

**Pull request merged**